### PR TITLE
[clang-format] Fix alignment in presence of template functions

### LINKED ISSUE #68102: [clang-format] Missing indention of an aligned template function call
With `AlignConsecutiveAssignment: true` and `ColumnLimit: 60` the Code
```c++
SomeName = Foo;
X        = func<Type, Type>(loong,
                            arrrrrrrrrrg);
```
gets formatted as
```c++
SomeName = Foo;
X        = func<Type, Type>(loong,
                     arrrrrrrrrrg);
```
which is obviously missing the indent of the `=`. It has to be a template function.

### PR DESCRIPTION
**Pull request by HazardyKnusperkeks**
Fixes #68102.

### CODE REVIEW

File: clang/unittests/Format/FormatTest.cpp [hunk 1]
```diff
@@ -18525,12 +18525,17 @@ TEST_F(FormatTest, AlignConsecutiveAssignments) {
                "                     a_longer_name_for_wrap}};",
                Alignment);
 
-  Alignment.ColumnLimit = 60;
+  Alignment = getLLVMStyleWithColumns(60);
+  Alignment.AlignConsecutiveAssignments.Enabled = true;
```
Comment (owenca) [inline clang/unittests/Format/FormatTest.cpp:18529]: Is this necessary? If `AlignConsecutiveAssignments` is already on, you can use `EXPECT_TRUE` instead.

**Comment by owenca** [comment]
Can you add an issue to show the incorrect format before this patch?

**Review by mydeveloperday — APPROVED** [review]

File: clang/unittests/Format/FormatTest.cpp (hunk 1 above)
Comment (owenca) [reply clang/unittests/Format/FormatTest.cpp:18529]: Nvm.

**Review by owenca — APPROVED** [review]

**Comment by HazardyKnusperkeks** [comment]
> Can you add an issue to show the incorrect format before this patch?

Done.

**Pull request merged**